taskvine/src/worker/vine_cache.c

```diff
@@ -228,6 +228,32 @@ int unpack_or_rename_target( struct cache_file *f, const char *transfer_path, co
 		debug(D_VINE,"unpacking %s to %s via command %s",transfer_path,cache_path,command);
 		unix_result = system(command);
 		free(command);
+	} else if(f-> flags & VINE_PONHO_UNPACK){
+		command = string_format("poncho_package_run -k -r -e %s --unpack-to %s", trasnfer_path, cache_path);
```

Comment:
Don't forget to free the command.

Reply:
Actually, you could just make this a single command line string:
`poncho_package_run ... && activate && set_env`
since it all needs to run to completion anyway..